### PR TITLE
Solved the verification Bug and migrated the test

### PR DESCRIPTION
**Pull request by Zartaj0**
This PR contains changes for https://github.com/ethereum-push-notification-service/push-smart-contracts/issues/253

### CODE REVIEW

**Review by 0xNilesh — CHANGES_REQUESTED** [review]
Few pointers:
1> can you include test cases for checking verifiedBy variable change in the fn (if correct address is populating in it or not)
2> if you can create separate files for verifyChannel and unverifyChannel, and even for getChannelVerification in this ChannelVerification directory, it'd be better
3> Ig tests for batchVerification will come in admin related test suits, if that's the case, then it's alright

**Comment by Zartaj0** [comment]
> Few pointers: 1> can you include test cases for checking verifiedBy variable change in the fn (if correct address is populating in it or not) 2> if you can create separate files for verifyChannel and unverifyChannel, and even for getChannelVerification in this ChannelVerification directory, it'd be better 3> Ig tests for batchVerification will come in admin related test suits, if that's the case, then it's alright

1. Good catch, done the changes. 
2. Made separate folders for both `verify` and `unverify`. I think `getChannelVerification` doesn't require a separate file as it is a view function only, which is being tested in both the mentioned test files. 
3. Admin Action test files are mainly for the independent setter functions, that don't come under any other functionality ending up being untested.

**Review by 0xNilesh — APPROVED** [review]
LGTM

File: test/PushCore/unit_tests/ChannelVerificationUnverification/ChannelVerification/ChannelVerification.t.sol [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        address Alice_verifiedByAfter = _getVerifiedBy(actor.alice_channel_owner);
+        assertEq(Alice_verifiedByAfter, actor.admin);
+    }
+
+    function test_RevertWhen_APrimaryVerifiedChannelVerifiesAnotherPrimaryVerifiedChannel() external {
+        // it should REVERT- not allowing downgrade primary verified to secondary
+
+        changePrank(actor.admin);
+        coreProxy.verifyChannel(actor.bob_channel_owner);
+        changePrank(actor.admin);
```
Comment (zaryab2000) [inline test/PushCore/unit_tests/ChannelVerificationUnverification/ChannelVerification/ChannelVerification.t.sol]: This line might be redundant. @Zartaj0

File: test/PushCore/unit_tests/ChannelVerifyTag/ChannelUnverification/ChannelUnverification.t.sol [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+        address Alice_verifiedBy = _getVerifiedBy(actor.alice_channel_owner);
+        assertEq(Alice_verifiedBy, address(0));
+    }
+
+    function test_RevertWhen_AChannelUnverifies_ASecondaryVerifiedChannel_VerifiedByAnotherChannel() external {
+        // it should REVERT
+        changePrank(actor.admin);
+        coreProxy.verifyChannel(actor.bob_channel_owner);
+        changePrank(actor.admin);
```
Comment (zaryab2000) [inline test/PushCore/unit_tests/ChannelVerifyTag/ChannelUnverification/ChannelUnverification.t.sol:76]: This line might be redundant @Zartaj0

**Comment by zaryab2000** [comment]
Couple of pointers found in the review of this PR:

1. The tree files need to be more specific and detailed for every test. For instance:

- The current tree files for verifications, doesn't include the event emission tests at all.
- Verify channel tree files doesn't specifically include the **_onlyActivatedChannels_** modifier related tests
- _As a result, these test cases are also not visible in the actual test files_
**Note:** Devote more time in designing/documenting tree files to ensure no branch is missing _(even for instances where the test is supposed to pass)_

2. Naming conventions of the tests can be made more readable for lengthy test names. For instances

Use `test_WhenChecked_TheVerificationStatusFor_AdminOrZeroAddress` instead of `test_WhenCheckedTheVerificationStatusForAdminOrZeroAddress`

@Zartaj0

**Review by zaryab2000 — CHANGES_REQUESTED** [review]

**Review by zaryab2000 — APPROVED** [review]

**Pull request merged**